hazelcast-client/src/main/java/com/hazelcast/client/cache/impl/AbstractClientInternalCacheProxy.java

```diff
@@ -319,10 +333,8 @@ protected void onRemoveAsyncInternal(Data keyData, ClientDelegatingFuture future
                 expiryPolicyData, completionId);
         ClientInvocationFuture future = invoke(request, keyData, completionId);
         ClientDelegatingFuture<T> delegatingFuture = newDelegatingFuture(future, REPLACE_RESPONSE_DECODER);
-        if (async) {
-            ExecutionCallback callback = statisticsEnabled ? new OnReplaceCallback(startNanos, false) : null;
-            onReplaceInternalAsync(keyData, newValueData, newValue, delegatingFuture, callback);
-        }
+        ExecutionCallback callback = async && statisticsEnabled ? new OnReplaceCallback(startNanos, false) : null;
+        onReplaceInternalAsync(keyData, newValueData, newValue, delegatingFuture, callback);
```

Comment:
We always have to call `onReplaceInternalAsync()`, otherwise the Near Cache proxy cannot add it's invalidation callback wrapper. All methods can handle a `null` callback fine, so it's just created if `async` and `statisticsEnabled` are both `true`.